pycbc/types/timeseries.py

```diff
@@ -251,51 +252,64 @@ def at_time(self, time, nearest_sample=False, interpolate=None):
         interpolate: str, None
             Return the interpolated value of the time series. Choices
             are simple linear or quadratic interpolation.
+        extrapolate: str or float, None
+            Value to return if time is outsidde the range of the vector or
+            method of extrapolating the value.
         """
+        if nearest_sample:
+            time = time + self.delta_t / 2.0
+        vtime = _numpy.array(time, ndmin=1)
+
+        fill_value = None
+        keep_idx = None
+        size = len(vtime)
+        if extrapolate is not None:
+            if _numpy.isscalar(extrapolate) and _numpy.isreal(extrapolate):
+                fill_value = extrapolate
+                facl = facr = 0
+                if interpolate == 'quadratic':
+                    facl = facr = 1.1
+                elif interpolate == 'linear':
+                    facl, facr = 0.1, 1.1
+
+                left = (vtime >= self.start_time + self.delta_t * facl)
+                right = (vtime < self.end_time - self.delta_t * facr)
+                keep_idx = _numpy.where(left & right)[0]
+                vtime = vtime[keep_idx]
+            else:
+                raise ValueError("Unsuported extrapolate: %s" % extrapolate)
+
+        fi = (vtime - float(self.start_time))*self.sample_rate
+        i = _numpy.asarray(_numpy.floor(fi)).astype(int)
+        di = fi - i
+
         if interpolate == 'linear':
-            i = (time - float(self.start_time))*self.sample_rate
-            di = i - int(i)
-            i = int(i)
             a = self[i]
             b = self[i+1]
-            return a + (b - a) * di
+            ans = a + (b - a) * di
         elif interpolate == 'quadratic':
-            ir = (time - float(self.start_time))*self.sample_rate
-            i = _numpy.floor(_numpy.asarray(ir)).astype(int)
-            di = ir - i
             c = self.data[i]
             xr = self.data[i + 1] - c
             xl = self.data[i - 1] - c
             a = 0.5 * (xr + xl)
             b = 0.5 * (xr - xl)
             ans = a * di**2.0 + b * di + c
-            return ans
-
-        if nearest_sample:
-            time += self.delta_t / 2.0
-        return self[int((time-self.start_time)*self.sample_rate)]
+        else:
```

Comment:
Jus making sure I know what this is doing, if no interpolate option is given then it will return the value of the closest previous point in the time series?

Reply:
Yes, that's the same behavior as before, so there should be no change change in output.